### PR TITLE
doctest pt 2

### PR DESCRIPTION
**Pull request by bearpelican**
nbtest side of things.
Continuation from this PR: https://github.com/fastai/fastai/pull/1634
Forum discussion: https://forums.fast.ai/t/improving-expanding-functional-tests/32929/164

Todo: 
update manifest
Test complete pipeline and handle multiple args for this_tests

@Benudek @stas00

### CODE REVIEW

File: fastai/gen_doc/doctest.py [hunk 1]
```diff
@@ -11,7 +14,7 @@ def this_tests(testedapi):
         prev_frame = inspect.currentframe().f_back.f_back
         pathfilename, line_number, test_function_name, lines, index = inspect.getframeinfo(prev_frame)
         lineno_parentfunc, parent_func = get_parent_func(line_number, get_lines(pathfilename))
-        list_test = [{'file': basename(pathfilename), 'test': test_function_name , 'line': str(lineno_parentfunc)}]
+        list_test = [{'file': relative_test_path(pathfilename), 'test': test_function_name , 'line': lineno_parentfunc}]
```
Comment (bearpelican) [inline fastai/gen_doc/doctest.py]: @Benudek 'file' now includes the 'tests' folder along with the name.

This was easier for me to link the github sourcecode and pytest command. I can change it back if you see any problems.

File: tests/conftest.py [hunk 2]
```diff
@@ -36,14 +36,14 @@ def pytest_collection_modifyitems(config, items):
 
 @pytest.fixture(scope="session", autouse=True)
 def start_doctest_collector(request):
-    matching = [s for s in set(sys.argv) if re.match(r'.*test_\w+\.py',s)]
-    if not matching: request.addfinalizer(stop_doctest_collector)
+#     matching = [s for s in set(sys.argv) if re.match(r'.*test_\w+\.py',s)]
```
Comment (stas00) [inline tests/conftest.py]: this should be removed

File: tests/test_vision_data.py [hunk 3]
```diff
@@ -239,120 +239,120 @@ def test_points():
...
-    im   = PIL.Image.open(f).convert("I") # so that the 16bit values are preserved as integers
-    vmax = pil2tensor(im,np.int).data.numpy().max()
-    assert(vmax>255)
-
-def test_vision_pil2tensor_numpy():
-    "assert that the two arrays contains the same values"
-    arr  = np.random.rand(16,16,3)
-    diff = np.sort( pil2tensor(arr,np.float).data.numpy().flatten() ) - np.sort(arr.flatten())
-    assert( np.sum(diff==0)==len(arr.flatten()) )
+# def test_coco():
```
Comment (stas00) [inline tests/test_vision_data.py]: these tests shouldn't be removed.

File: fastai/gen_doc/nbdoc.py [hunk 4]
```diff
@@ -256,22 +246,6 @@ def get_module_toc(mod_name):
                 tabmat += f'  - [{name}](#{name})\n'
     display(Markdown(tabmat))
 
-def get_class_toc(mod_name:str, cls_name:str):
```
Comment (bearpelican) [inline fastai/gen_doc/nbdoc.py]: just removing old code

File: tests/conftest.py (hunk 2 above)
Comment (bearpelican) [reply tests/conftest.py]: Haha realized right after I opened the PR. Your review is too fast for me!

**Comment by bearpelican** [comment]
Sorry tagged too early 😭 Should be stable now

**Comment by stas00** [comment]
It's in your domain, @bearpelican, so I trust it's how you always deliver. But otherwise looks great.

Thank you!

**Pull request merged**